x-pack/plugins/observability/public/embeddable/slo/slo_embeddable.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Subscription } from 'rxjs';
+
+import {
+  Embeddable as AbstractEmbeddable,
+  EmbeddableOutput,
+  IContainer,
+} from '@kbn/embeddable-plugin/public';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import dateMath from '@kbn/datemath';
+
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { SloOverview } from './slo_overview';
+import type { SloEmbeddableDeps, SloEmbeddableInput } from './types';
+
+export const SLO_EMBEDDABLE = 'SLO_EMBEDDABLE';
+
+function datemathToEpochMillis(
+  value: string,
+  round: 'down' | 'up' = 'down',
+  forceNow?: Date
+): number | null {
+  const parsedValue = dateMath.parse(value, { roundUp: round === 'up', forceNow });
+  if (!parsedValue || !parsedValue.isValid()) {
+    return null;
+  }
+  return parsedValue.valueOf();
+}
+
+export class SLOEmbeddable extends AbstractEmbeddable<SloEmbeddableInput, EmbeddableOutput> {
+  public readonly type = SLO_EMBEDDABLE;
+  private subscription: Subscription;
+  private node?: HTMLElement;
+
+  constructor(
+    private readonly deps: SloEmbeddableDeps,
+    initialInput: SloEmbeddableInput,
+    parent?: IContainer
+  ) {
+    super(initialInput, {}, parent);
+
+    this.subscription = new Subscription();
+    this.subscription.add(this.getInput$().subscribe(() => this.reload()));
+  }
+
+  public render(node: HTMLElement) {
+    this.node = node;
+    const startTimestamp = datemathToEpochMillis(this.input.timeRange!.from);
+    const endTimestamp = datemathToEpochMillis(this.input.timeRange!.to, 'up');
+    const { sloId, sloInstanceId } = this.getInput();
+    const queryClient = new QueryClient();
+
+    const I18nContext = this.deps.i18n.Context;
+    ReactDOM.render(
+      <I18nContext>
+        <KibanaContextProvider services={this.deps}>
+          <QueryClientProvider client={queryClient}>
+            <SloOverview
+              sloId={sloId}
+              sloInstanceId={sloInstanceId}
+              startTime={startTimestamp}
+              endTime={endTimestamp}
```

Comment:
Yep this works fine, but doesn't look good. I didn't know how to listen on the refresh button click, and that's why I did it like this.

I found [this](https://github.com/elastic/kibana/blob/main/src/plugins/embeddable/common/types.ts#L35). I think it will do the job. Let me try

Reply:
It works with the `lastReloadRequestTime`. It causes the SLO widget to refresh. 

@kdelemme Does it look better now? It is definitely cleaner this way!